servers/store/src/main/proto/table.proto

```diff
@@ -22,17 +22,22 @@ option java_generate_equals_and_hash = true;
 
 message Contents {
   oneof object_type {
-    IcebergTable iceberg_table = 1;
-    SqlView sql_view = 2;
-    DeltaLakeTable delta_lake_table = 3;
+    IcebergTableMetadata iceberg_table_metadata = 1;
+    IcebergSnapshot iceberg_snapshot = 2;
+    SqlView sql_view = 3;
+    DeltaLakeTable delta_lake_table = 4;
   }
-  string id = 6;
+  string id = 5;
 }
 
-message IcebergTable {
+message IcebergTableMetadata {
```

Comment:
Note: renaming the protobuf type to make clear that it's not a 1:1 to the REST type IcebergTable